### PR TITLE
Change error in case not-found Tool requested

### PR DESCRIPTION
**Pull request by mwichmann**
This is basically a cosmetic change:

The previous error when requesting a Tool module that isn't found is a lengthy traceback and an error msg "`SConsEnvironmentError: No module named foo:`" modeled on Python's infamous error on failing to import. This isn't very communicative, so the error is changed to a `UserError`, which limits the traceback to just the SConstruct line, and the message is updated and now includes the tool dirs searched.  There's a model change here: we'll be telling the user that putting an unknown tool (or a tool in an incorrectly specified path) into an SConscript is an error in the SConscript, thus UserError, and not an error in environment.

The two affected unit tests are updated to expect `UserError` in this circumstance. Along the way, an existing technique from one of the unit tests is moved over to the other, allowing the elimination of a bare raise which was flagged by pylint (marked TODO in the file).  The same change is then applied to a third unit test, unrelated but marked with the same TODO.

Unsure if anyone depends on the previous behavior...

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [X] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

**Comment by bdbaddog** [comment]
No test verifying string for new exception output string?

**Comment by mwichmann** [comment]
No, I didn't do anything for that, just updated the couple of unit tests that touched this, and they didn't look for message, only for "correct exception".  I'll have another glance later...

File: CHANGES.txt [hunk 1]
```diff
@@ -35,6 +35,8 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
       OverrideEnvironment. Enable env.setdefault() method, add tests.
     - Raise an error if an option (not otherwise consumed) is used which
       looks like an abbreviation of one one added by AddOption. (#3653)
+    - Change error (to UserError) and error message in case a requested
```
Comment (bdbaddog) [inline CHANGES.txt]: When a Tool() is requested but not found,  a UserError will new be thrown and potentially and error message if requested.  

Lead with Tool...

**Comment by bdbaddog** [comment]
Test change looks good, just a sugguestion to change the CHANGES.txt blurb to indicate up from it's about loading tools. Also.. can you add a blurb to RELEASE.rst ?

File: CHANGES.txt (hunk 1 above)
Comment (mwichmann) [reply CHANGES.txt]: Done.

**Pull request merged**